scripts/sct_run_batch.py

```diff
@@ -408,6 +451,15 @@ def reset_streams():
     print('To open the Quality Control (QC) report on a web-browser, run the following:\n'
           '{} {}/index.html'.format(open_cmd, path_qc))
 
+    if args.zip:
+        path_tmp = os.path.join(tempfile.mkdtemp(), 'sct_run_batch_{}'.format(time.strftime('%Y%m%d%H%M%S')))
```

Comment:
There is a better way than doing the copytree then zipping, the zip can be made on the fly.
@Drulex maybe you can help out?

Reply:
Probably something like this:
```
if args.zip:
        fname_zip = os.path.join(path_output, 'sct_run_batch_{}.zip'.format(time.strftime('%Y%m%d%H%M%S')))
        with zipfile.ZipFile(fname_zip, "w") as z:
            for folder in [path_log, path_qc, path_results]:
                for root, _, files in os.walk(folder): 
                    for file in files:
                        fpath = os.path.join(root, file)
                        if os.path.isfile(fpath):
                            z.write(fpath, os.path.basename(fpath))
```
(not tested)

Reply:
^ this code does not zip according to the directory tree of the native folder (everything is flattened), hence the added `utils.zipdir()`.

but i will look into making it on the fly if i have a few minutes to spare

Reply:
fixed in 0407fc2fc5b2782080d62e801e42d94c76996279